### PR TITLE
[ALLUXIO-1871] Change alluxio-start.sh to follow Google bash style guide

### PR DESCRIPTION
**Pull request by peisun1115**
Mostly to address Jiri's comment to follow Google bash style guide or Alluxio's bash convection.

### CODE REVIEW

**Comment by ifcharming** [comment]
ok to test

**Comment by gpang** [comment]
Jenkins, add to whitelist

File: bin/alluxio-start.sh [hunk 1]
```diff
@@ -179,112 +181,112 @@ run_safe() {
...
-    ;;
-  restart_worker)
-    restart_worker
-    ;;
-  restart_workers)
-    ${LAUNCHER} ${BIN}/alluxio-workers.sh ${BIN}/alluxio-start.sh restart_worker
-    ;;
-  *)
-    echo "Error: Invalid WHAT: ${WHAT}"
+      ${LAUNCHER} "${BIN}/alluxio-workers.sh" "${BIN}"/alluxio-start.sh "worker" "${MOPT}"
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: nit: I prefer `"${BIN}/alluxio-start.sh" over `"${BIN}"/alluxio-start.sh`

File: bin/alluxio-start.sh [hunk 2]
```diff
@@ -179,112 +181,112 @@ run_safe() {
...
+      start_master "${FORMAT}"
+      ;;
+    worker)
+      start_worker "${MOPT}"
+      ;;
+    safe)
+      run_safe
+      ;;
+    workers)
+      ${LAUNCHER} "${BIN}"/alluxio-workers.sh "${BIN}"/alluxio-start.sh worker "${MOPT}" \
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: ditto

File: bin/alluxio-start.sh [hunk 3]
```diff
@@ -179,112 +181,112 @@ run_safe() {
...
+       "${ALLUXIO_MASTER_ADDRESS}"
+      ;;
+    restart_worker)
+      restart_worker
+      ;;
+    restart_workers)
+      ${LAUNCHER} "${BIN}"/alluxio-workers.sh "${BIN}"/alluxio-start.sh restart_worker
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: ditto

File: bin/alluxio-start.sh [hunk 4]
```diff
@@ -179,112 +181,112 @@ run_safe() {
...
 
-if [[ "${wait}" ]]; then
-  wait
-fi
+  if [[ "${wait}" ]]; then
+    wait
+  fi
+}
+
+main "$@"
```
Comment (jsimsa) [inline bin/alluxio-start.sh]: end file with a newline

**Comment by jsimsa** [comment]
@peisun1115 thank you for the cleanup ... as you correctly pointed out, some of the existing shell scripts do not follow the Google shell style, I try to incrementally improve the situation by fixing things related to my PRs and comments on other people PRs

File: bin/alluxio-start.sh (hunk 4 above)
Comment (peisun1115) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 3 above)
Comment (peisun1115) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 2 above)
Comment (peisun1115) [inline bin/alluxio-start.sh]: done

File: bin/alluxio-start.sh (hunk 1 above)
Comment (peisun1115) [inline bin/alluxio-start.sh]: done

**Comment by peisun1115** [comment]
Jiri, I have addressed your comments. PTAL. Thanks

**Comment by jsimsa** [comment]
LGTM

File: bin/alluxio-start.sh [hunk 5]
```diff
@@ -179,112 +181,113 @@ run_safe() {
...
+        wait="true"
+        ;;
+      *)
+        echo -e "${USAGE}"
+        exit 1
+        ;;
+    esac
+  done
+
+  shift $((OPTIND-1))
```
Comment (ifcharming) [inline bin/alluxio-start.sh]: nit: shift $((${OPTIND} - 1)) is preferred.

Per google style guide:
# Prefer this:
addition=$((${X} + ${Y}))
substitution="${string/#foo/bar}"

**Comment by ifcharming** [comment]
Thanks for the cleanup! I left a minor comment, otherwise LGTM

File: bin/alluxio-start.sh (hunk 5 above)
Comment (peisun1115) [inline bin/alluxio-start.sh]: done

**Comment by ifcharming** [comment]
LGTM

**Pull request merged**